packages/core/src/extensions/FormattingToolbar/FormattingToolbarFactoryTypes.ts

```diff
@@ -23,7 +23,7 @@ export type FormattingToolbarDynamicParams = {
   // BlockContentType is mostly used to set a block's type, so the attr field is optional as block content types have
   // default values for attributes. However, it means that a block type's attributes field will never be undefined due to
   // these default values, which the Required type enforces.
-  activeBlockType: Required<BlockContentType>;
+  activeBlockType: Required<BlockContent>;
```

Comment:
Hmm, I'm wondering about the comment above. If I understand it correctly, then I think we're using types a bit wrong here.

I understand:
- `BlockContent` marks `attrs` as optional, because it's mostly used for setting
- `Required<BlockContent>` is the actual type, because the attrs should always be set.

Based on this, I think we should revise this, and make sure the type `BlockContent` is always "complete". e.g.: it sounds like we both think a `heading` without a `level` is not a valid type. Then this should be reflected in the HeadingBlockType. 

If you want to make it possible to create a `HeadingBlockType` without specifying a `level`, then that concern should be handled somewhere that has that specific responsibility (to create a `HeadingBlockType` with a sensible default level. We should not be passing semi-complete types around in the rest of the code base (and then at some point mark them `Required` when needed.

Reply:
I just realized that since we can pass in either `BlockContent` or `BlockContent["name"]`, having the attributes field be optional is totally redundant. E.g. if we want to make a heading with default attributes we can just call `BNSetContentType(pos, "heading")` so giving users the option to do the same thing with `BNSetContentType(pos, { name: "heading" })` is pretty pointless.

Reply:
Another update, the original idea I had was flawed anyway since the `attrs` field actually needs to be `Partial`, not optional, when setting a block type. Getting a block type should still always return the full `attrs`. The only way I've found of doing this is making a second, settable type with `Partial<Attrs>` for `Paragraph`s, `Heading`s, and `ListItem`s, then declaring a new type like `type SettableBlockContent = SettableParagraph | SettableHeading | SettableListItem`, but this results in a huge amount of duplicated code. I can't do something similar after already declaring `type BlockContent = Paragraph | Heading | ListItem` as it will allow type names and attributes to be mixed and matched (e.g. it won't warn you if you try pass `{ name: "paragraph", attrs: { level: "1" }}`).